tap_snapchat_ads/__init__.py

```diff
@@ -31,10 +33,20 @@ def do_discover():
 def main():
 
     parsed_args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
+    
+    request_timeout = parsed_args.config.get('request_timeout')
+    
+    # if request_timeout is other than 0, "0" or "" then use request_timeout
+    if request_timeout and float(request_timeout):
+        request_timeout = float(request_timeout)
+    else: # If value is 0, "0" or "" then set the default which is 300 seconds.
+        request_timeout = REQUEST_TIMEOUT
+        
 
     with SnapchatClient(parsed_args.config['client_id'],
                         parsed_args.config['client_secret'],
                         parsed_args.config['refresh_token'],
+                        request_timeout,
```

Comment:
Pass `parsed_args.config.get('request_timeout')` instead of `request_timeout` in object initilization and move calcuation to client.py to keep consistensy.

Reply:
Passed parsed_args.config.get('request_timeout') insted of request_timeout in SanapchatClient object